Mathlib/CategoryTheory/Grothendieck.lean

```diff
@@ -369,6 +427,87 @@ def pre (G : D ⥤ C) : Grothendieck (G ⋙ F) ⥤ Grothendieck F where
   map_id X := Grothendieck.ext _ _ (G.map_id _) (by simp)
   map_comp f g := Grothendieck.ext _ _ (G.map_comp _ _) (by simp)
 
+@[simp]
+theorem pre_id : pre F (𝟭 C) = 𝟭 _ := rfl
+
+/--
+An natural isomorphism between functors `G ≅ H` induces a natural isomorphism between the canonical
+morphism `pre F G` and `pre F H`, up to composition with
+`Grothendieck (G ⋙ F) ⥤ Grothendieck (H ⋙ F)`.
+-/
+def preNatIso {G H : D ⥤ C} (α : G ≅ H) :
+    pre F G ≅ map (whiskerRight α.hom F) ⋙ (pre F H) :=
+  NatIso.ofComponents
+    (fun X => (transportIso ⟨G.obj X.base, X.fiber⟩ (α.app X.base)).symm)
+    (fun f => by fapply Grothendieck.ext <;> simp)
+
+/--
+Given an equivalence of categories `G`, `preInv _ G` is the (weak) inverse of the `pre _ G.functor`.
+-/
+def preInv (G : D ≌ C) : Grothendieck F ⥤ Grothendieck (G.functor ⋙ F) :=
+  map (whiskerRight G.counitInv F) ⋙ Grothendieck.pre (G.functor ⋙ F) G.inverse
+
+variable {F} in
+lemma pre_comp_map (G: D ⥤ C) {H : C ⥤ Cat} (α : F ⟶ H) :
+    pre F G ⋙ map α = map (whiskerLeft G α) ⋙ pre H G := rfl
+
+variable {E : Type*} [Category E] in
+@[simp]
+lemma pre_comp (G : D ⥤ C) (H : E ⥤ D) : pre F (H ⋙ G) = pre (G ⋙ F) H ⋙ pre F G := rfl
+
+/--
+Let `G` be an equivalence of categories. The functor induced via `pre` by `G.functor ⋙ G.inverse`
+is naturally isomorphic to the functor induced via `map` by a whiskered version of `G`'s inverse
+unit.
+-/
+protected def preUnitIso (G : D ≌ C) :
+    map (whiskerRight G.unitInv _) ≅ pre (G.functor ⋙ F) (G.functor ⋙ G.inverse) :=
+  preNatIso _ G.unitIso.symm |>.symm
+
+/--
+Given a functor `F : C ⥤ Cat` and an equivalence of categories `G : D ≌ C`, the functor
+`pre F G.functor` is an equivalence between `Grothendieck (G.functor ⋙ F)` and `Grothendieck F`.
+-/
+def preEquivalence (G : D ≌ C) : Grothendieck (G.functor ⋙ F) ≌ Grothendieck F := by
+  refine ⟨(pre F G.functor), (preInv F G), ?_, ?_, ?_⟩
```

Comment:
```suggestion
def preEquivalence (G : D ≌ C) : Grothendieck (G.functor ⋙ F) ≌ Grothendieck F where
  functor := pre F G.functor
  inverse := preInv F G
```
Also, the unit and counit isos should not be constructed using tactics. This would create issues when we want to unfold these definitions.

Reply:
Now the construction of `unitIso` begins with a `refine` tactic and its only hole is propositional (`eqToIso ?_`). I find it more readable this way, compared to directly filling the hole.

As far as I know, using `refine` should be okay as it just helps me build an expression in multiple steps. Is it fine this way or should I really use term mode here?